Tests/Marketplace/search_and_install_packs.py

```diff
@@ -457,7 +464,11 @@ def search_pack_and_its_dependencies(client: demisto_client,
 
         lock.acquire()
         if one_pack_and_its_dependencies_in_batch:
-            batch_packs_install_request_body.append(current_packs_to_install)      # type:ignore[union-attr]
+            pack_and_its_dependencies = []
+            for pack in current_packs_to_install:
```

Comment:
A proposal for a different approach:
1. Remove duplicates from pack_and_its_dependencies (e.g. with a set, or a dict)
2. Aggregate the distinct ones

This way to do `o(n)` instead of `o(n^2)`